src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -245,11 +270,37 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state)
             }
 
             SetProxyOptions(state, requestHandle);
+            SetRequestHandleDecompressionOptions(requestHandle);
             // TODO: Handle headers and other options
 
             return requestHandle;
         }
 
+        private void SetRequestHandleDecompressionOptions(SafeCurlHandle requestHandle)
+        {
+            var curlEncoding = new StringBuilder();
+            if (AutomaticDecompression != DecompressionMethods.None)
+            {
+                var needComma = false;
+                if ((AutomaticDecompression & DecompressionMethods.GZip) != 0)
+                {
+                    curlEncoding.Append(EncodingNameGzip);
+                    needComma = true;
+                }
+
+                if ((AutomaticDecompression & DecompressionMethods.Deflate) != 0)
+                {
+                    if (needComma)
+                    {
+                        curlEncoding.Append(',');
+                    }
+
+                    curlEncoding.Append(EncodingNameDeflate);
+                }
+                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_ACCEPTENCODING, curlEncoding.ToString());
+            }
+        }
+
```

Comment:
Actually, disregard my comment... Since `EncodingNameGzip` and `EncodingNameDeflate` are already constants, my suggestion is really no different than what @stephentoub suggests.